src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/cost/RouteCostCalculator.java

```diff
@@ -0,0 +1,81 @@
+package org.opentripplanner.routing.algorithm.raptoradapter.transit.cost;
+
+import java.util.Map;
+import javax.annotation.Nonnull;
+import org.opentripplanner.routing.algorithm.raptoradapter.transit.TripSchedule;
+import org.opentripplanner.transit.model.framework.FeedScopedId;
+import org.opentripplanner.transit.raptor.api.transit.CostCalculator;
+import org.opentripplanner.transit.raptor.api.transit.RaptorTransfer;
+import org.opentripplanner.transit.raptor.api.transit.RaptorTransferConstraint;
+
+public class RouteCostCalculator<T extends DefaultTripSchedule> implements CostCalculator<T> {
+
+  private final CostCalculator delegate;
+  private final Map<FeedScopedId, Integer> routePenalties;
+
+  public RouteCostCalculator(
+    @Nonnull CostCalculator<T> delegate,
+    @Nonnull Map<FeedScopedId, Integer> routePenalties
+  ) {
+    this.routePenalties = routePenalties;
+    this.delegate = delegate;
+  }
+
+  @Override
+  public int boardingCost(
+    boolean firstBoarding,
+    int prevArrivalTime,
+    int boardStop,
+    int boardTime,
+    T trip,
+    RaptorTransferConstraint transferConstraints
+  ) {
+    return delegate.boardingCost(
+      firstBoarding,
+      prevArrivalTime,
+      boardStop,
+      boardTime,
+      trip,
+      transferConstraints
+    );
+  }
+
+  @Override
+  public int onTripRelativeRidingCost(int boardTime, T tripScheduledBoarded) {
+    return delegate.onTripRelativeRidingCost(boardTime, tripScheduledBoarded);
+  }
+
+  @Override
+  public int transitArrivalCost(
+    int boardCost,
+    int alightSlack,
+    int transitTime,
+    T trip,
+    int toStop
+  ) {
+    int defaultCost = delegate.transitArrivalCost(
+      boardCost,
+      alightSlack,
+      transitTime,
+      trip,
+      toStop
+    );
+    int routeReluctanceCost = routePenalties.getOrDefault(trip.routeId(), 0);
```

Comment:
```suggestion
    int routeReluctanceCost = routePenalties.getOrDefault(trip.routeId(), ZERO_COST);
```

Reply:
I think you should add this cost at board time, not arrival time - technically that is more correct. I am not sure if there are more boarding events than alighting events. I should probably know  - so I will test that later. A trip is boarded for every stop reached along the route, possibly by more than one path, and alighted n times (all pareto optimal paths) at all sops after that.